services/src/main/java/org/keycloak/services/ParValidationService.java

```diff
@@ -0,0 +1,159 @@
+package org.keycloak.services;
+
+import org.jboss.resteasy.spi.HttpRequest;
+import org.keycloak.OAuth2Constants;
+import org.keycloak.OAuthErrorException;
+import org.keycloak.events.Details;
+import org.keycloak.events.Errors;
+import org.keycloak.events.EventBuilder;
+import org.keycloak.models.ClientModel;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.protocol.oidc.OIDCLoginProtocol;
+import org.keycloak.protocol.oidc.TokenManager;
+import org.keycloak.protocol.oidc.endpoints.request.AuthorizationEndpointRequest;
+import org.keycloak.protocol.oidc.endpoints.request.AuthorizationEndpointRequestParserProcessor;
+import org.keycloak.protocol.oidc.utils.OIDCRedirectUriBuilder;
+import org.keycloak.protocol.oidc.utils.OIDCResponseMode;
+import org.keycloak.protocol.oidc.utils.OIDCResponseType;
+import org.keycloak.protocol.oidc.utils.RedirectUtils;
+import org.keycloak.representations.idm.OAuth2ErrorRepresentation;
+import org.keycloak.util.TokenUtil;
+
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+public class ParValidationService {
+
+    private KeycloakSession session;
+    private EventBuilder event;
+    private AuthorizationEndpointRequest authorizationRequest;
+    private OIDCResponseMode parsedResponseMode;
+    private String redirectUri;
+
+    public ParValidationService(KeycloakSession session, EventBuilder event) {
+        this.session = session;
+        this.event = event;
+    }
+
+    public Response validateParRequest(HttpRequest request, ClientModel clientModel) {
```

Comment:
According to https://datatracker.ietf.org/doc/html/draft-ietf-oauth-par#section-4 ,

>     3.  If the client has authentication credentials established with the
>       authorization server, reject the request if the authenticated
>       "client_id" does not match the "client_id" claim in the request
>       object.

It seems not to be implemented.